components/butterfly/src/server/mod.rs

```diff
@@ -622,17 +622,22 @@ impl Server {
         // NOTE: This sucks so much right here. Check out how we allocate no matter what, because
         // of just how the logic goes. The value of the trace is really high, though, so we suck it
         // for now.
-        let trace_member_id = member.id.clone();
+        let member_id = member.id.clone();
         let trace_incarnation = member.incarnation;
         let trace_health = health.clone();
         if self.member_list.insert(member, health) {
             trace_it!(
                 MEMBERSHIP: self,
                 TraceKind::MemberUpdate,
-                trace_member_id,
+                member_id,
                 trace_incarnation,
                 trace_health
             );
+
+            // Purge "heat" information for a member that's gone
```

Comment:
Again, I'm not sure this comment says anything the code doesn't convey quite clearly (ditto to the several other times this same comment is used).

What *is* confusing to me though is why we still `start_hot_rumor` even if we're purging. I'm not saying this is wrong (still trying to grok this code), but it seems strange enough that a comment about _that_ might be warranted.